### PR TITLE
bechmarks for width 8 as well as 10

### PR DESCRIPTION
**Pull request by gballet**

### CODE REVIEW

**Comment by gballet** [comment]
Seems to be a bit slower, the good news is that nodes with 256 children indeed seem to be faster.

Before:
```
goos: freebsd
goarch: amd64
pkg: github.com/gballet/go-verkle
cpu: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
BenchmarkCommit1kLeaves/insert-8         	      10	 111580384 ns/op	13560201 B/op	   11889 allocs/op
BenchmarkCommit1kLeaves/insertOrdered-8  	       8	 138136091 ns/op	13706082 B/op	   14880 allocs/op
BenchmarkCommit10kLeaves/insert-8        	       1	1037862170 ns/op	58394352 B/op	  107861 allocs/op
BenchmarkCommit10kLeaves/insertOrdered-8 	       1	1176266096 ns/op	59294032 B/op	  126046 allocs/op
BenchmarkCommitFullNode-8                	      25	  53143327 ns/op	  324609 B/op	    7757 allocs/op
PASS
ok  	github.com/gballet/go-verkle	8.026s
?   	github.com/gballet/go-verkle/benchs	[no test files]
```

After:
```
goos: freebsd
goarch: amd64
pkg: github.com/gballet/go-verkle
cpu: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
BenchmarkCommitLeaves/insert/1000-leaves/10-width-8         	      10	 137389485 ns/op	13301428 B/op	   11759 allocs/op
BenchmarkCommitLeaves/insertOrdered/1000-leaves/10-width-8  	       5	 238257053 ns/op	13445438 B/op	   14713 allocs/op
BenchmarkCommitLeaves/insert/10000-leaves/10-width-8        	       1	1869996729 ns/op	58344160 B/op	  107876 allocs/op
BenchmarkCommitLeaves/insertOrdered/10000-leaves/10-width-8 	       1	1994591041 ns/op	59248912 B/op	  126092 allocs/op
BenchmarkCommitLeaves/insert/1000-leaves/8-width-8          	       6	 191363082 ns/op	13406385 B/op	   11822 allocs/op
BenchmarkCommitLeaves/insertOrdered/1000-leaves/8-width-8   	       5	 201077508 ns/op	13551766 B/op	   14801 allocs/op
BenchmarkCommitLeaves/insert/10000-leaves/8-width-8         	       1	1662443269 ns/op	58544512 B/op	  107929 allocs/op
BenchmarkCommitLeaves/insertOrdered/10000-leaves/8-width-8  	       1	1785683726 ns/op	59447776 B/op	  126184 allocs/op
BenchmarkCommitFullNode-8                                   	      18	  65228327 ns/op	  324591 B/op	    7757 allocs/op
PASS
ok  	github.com/gballet/go-verkle	17.796s
?   	github.com/gballet/go-verkle/benchs	[no test files]
```

**Comment by gballet** [comment]
`BenchmarkModifyLeaves` times out in 8-bit mode. Not including it in this PR.

File: tree_test.go [hunk 1]
```diff
@@ -351,38 +352,44 @@ func TestFlush1kLeaves(t *testing.T) {
...
+	benchmarkCommitFullNode(b, 8)
+}
+func benchmarkCommitFullNode(b *testing.B, width int) {
+	b.Run(fmt.Sprintf("width/%d", width), func(b *testing.B) {
+		nChildren := 1 << width
+		value := []byte("value")
+		keys := make([][]byte, nChildren)
+		for i := 0; i < nChildren; i++ {
+			key := make([]byte, 32)
+			binary.BigEndian.PutUint16(key[:2], uint16(i)<<6)
```
Comment (s1na) [inline tree_test.go:373]: This only works for width 10. For width 8 I imagine it has to be something like:

```go
key[0] = byte(i)
```

**Review by s1na — CHANGES_REQUESTED** [review]

**Comment by s1na** [comment]
I was wondering why the "before" and "after" numbers for width 10 vary so much, but I guess your computer had different loads. They're consistent for me locally, so no problemo.

**Comment by gballet** [comment]
Yeah, by reducing the load I get more reasonable numbers:

```
goos: freebsd
goarch: amd64
pkg: github.com/gballet/go-verkle
cpu: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
BenchmarkCommitLeaves/insert/leaves/1000/width/10-8         	      12	 117802986 ns/op	13403226 B/op	   11797 allocs/op
BenchmarkCommitLeaves/insertOrdered/leaves/1000/width/10-8  	       8	 126891514 ns/op	13547972 B/op	   14779 allocs/op
BenchmarkCommitLeaves/insert/leaves/10000/width/10-8        	       1	1015655839 ns/op	58544272 B/op	  107846 allocs/op
BenchmarkCommitLeaves/insertOrdered/leaves/10000/width/10-8 	       1	1226915229 ns/op	59443888 B/op	  126013 allocs/op
BenchmarkCommitLeaves/insert/leaves/1000/width/8-8          	      10	 100400444 ns/op	 3672309 B/op	   12368 allocs/op
BenchmarkCommitLeaves/insertOrdered/leaves/1000/width/8-8   	       7	 155170307 ns/op	 3807441 B/op	   15179 allocs/op
BenchmarkCommitLeaves/insert/leaves/10000/width/8-8         	       1	1296841268 ns/op	17404784 B/op	  107254 allocs/op
BenchmarkCommitLeaves/insertOrdered/leaves/10000/width/8-8  	       1	1348349594 ns/op	18221384 B/op	  124100 allocs/op
BenchmarkCommitFullNode/width/10-8                          	      21	  54620227 ns/op	  324558 B/op	    7757 allocs/op
BenchmarkCommitFullNode/width/8-8                           	      86	  14033790 ns/op	   81230 B/op	    1932 allocs/op
PASS
ok  	github.com/gballet/go-verkle	15.573s
```

**Review by s1na — APPROVED** [review]
LGTM

**Pull request merged**